persistent/Database/Persist/Sql/Run.hs

```diff
@@ -113,7 +113,7 @@ withSqlPool mkConn connCount f = withUnliftIO $ \u -> bracket
     (unliftIO u . f)
 
 createSqlPool
-    :: (MonadLogger m, MonadUnliftIO m, IsSqlBackend backend)
+    :: (MonadLogger m, MonadUnliftIO m, BackendCompatible SqlBackend backend)
```

Comment:
Hm, if these functions can return a `BackendCompatible SqlBackend backend => backend` then shouldn't the other `{with,create}{Postgresql,MySQL,etc}Pool` functions also be able to create that?

Reply:
Well, no, since those functions are actual implementations, they will create `SqlBackend`s.

The only thing I can think of that wouldn't just imply using `IsSqlBackend`, would be an extra argument somewhere in which the user of the function can give the following function which will be called before creating the `Pool`:
`BackendCompatible SqlBackend backend => SqlBackend -> backend`
Maybe with a `MonadIO m` so they can use IO to create their own `backend`.